display_list/dl_color.h

```diff
@@ -58,58 +77,111 @@ struct DlColor {
   static constexpr DlColor kCornflowerBlue()     {return DlColor(0xFF6495ED);};
   // clang-format on
 
-  constexpr bool isOpaque() const { return getAlpha() == 0xFF; }
-  constexpr bool isTransparent() const { return getAlpha() == 0; }
-
-  constexpr int getAlpha() const { return argb_ >> 24; }
-  constexpr int getRed() const { return (argb_ >> 16) & 0xFF; }
-  constexpr int getGreen() const { return (argb_ >> 8) & 0xFF; }
-  constexpr int getBlue() const { return argb_ & 0xFF; }
-
-  constexpr DlScalar getAlphaF() const { return toF(getAlpha()); }
-  constexpr DlScalar getRedF() const { return toF(getRed()); }
-  constexpr DlScalar getGreenF() const { return toF(getGreen()); }
-  constexpr DlScalar getBlueF() const { return toF(getBlue()); }
-
+  constexpr bool isOpaque() const { return alpha_ >= 1.f; }
+  constexpr bool isTransparent() const { return alpha_ <= 0.f; }
+
+  ///\deprecated Use floating point accessors to avoid data loss when using wide
```

Comment:
Is that a spurious `\`?

Reply:
My point about these accessors are whether or not they guarantee values in the range 0-255? They should probably have some doc comments talking about that. Also, the class should have a doc comment talking about its new, more complicated, behaviors and policies. It used to just be a uint32_t standin which didn't require much documentation (but maybe should have talked about the argb ordering at the time).

Not anything huge as we might expect for app developer facing docs, but enough so that graphics developers have a reference for where the color spaces are defined and how the floats and ints relate to each other.

Reply:
Nope, that's a doxygen directive: https://www.doxygen.nl/manual/commands.html#cmddeprecated

I added a class docstring that reiterates colorspaces and deprecation of integer data accessors.